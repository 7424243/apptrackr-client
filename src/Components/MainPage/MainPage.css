/* Classes */

.filter {
    margin-top: 20px;
    margin-bottom: 20px;
    font-size: large;
}

.main_header, .no_applications {
    text-align: center;
}

.section1 {
    text-align: center;
}

.section1_content {
    margin-top: 20px;
    margin-left: 20px;
    margin-right: 20px;
    padding: 15px;
}

.section2_content {
    padding: 15px;
    text-align: center;
}

.section3 {
    margin-left: 20px;
    margin-right: 20px;
}

.quote {
    font-size: large;
}

/* ids */

#filter {
    border-style: solid;
}