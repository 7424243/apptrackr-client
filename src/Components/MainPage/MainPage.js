import React, {Component} from 'react'
import SquareButton from '../SquareButton/SquareButton'
import Table from '../Table/Table'
import './MainPage.css'

class MainPage extends Component {
    render() {
        return (
            <div className='main_container'>
                <section className='section1'>
                    <div className='section1_content'>
                        <p>"If one advances confidently in the direction of his dreams,
                            and endeavors to live the life which he has imagined,
                            he will meet with a success unexpected in common hours." ~ Henry David Thoreau 
                        </p>
                    </div>
                </section>
                <section className='section2'>
                    <h2 className='main_header'>Job Applications</h2>
                    <div className='section2_content'>
                        <SquareButton 
                            content={'+'}
                            path={'/newapp'}
                        />
                        <div>
                            <label>Filter By: </label>
                            <select>
                                <option>All</option>
                                <option>Interested</option>
                                <option>Applied</option>
                                <option>Closed</option>
                            </select>
                        </div>
                    </div>
                </section>
                <section className='section3'>
                    <Table />
                </section>
            </div>
        )
    }
}

export default MainPage