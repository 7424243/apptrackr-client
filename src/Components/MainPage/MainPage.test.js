import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import MainPage from './MainPage'

describe('MainPage Component', () => {

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><MainPage/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter><MainPage/></BrowserRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    
})