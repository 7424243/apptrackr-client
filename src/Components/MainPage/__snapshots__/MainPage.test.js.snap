// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component renders the UI as expected 1`] = `
Array [
  <div
    className="main_container"
  >
    <section
      className="section1"
    >
      <div
        className="section1_content"
      >
        <p
          className="quote"
        >
          "If one advances confidently in the direction of his dreams, and endeavors to live the life which he has imagined, he will meet with a success unexpected in common hours." ~ Henry David Thoreau
        </p>
      </div>
    </section>
    <section
      className="section2"
    >
      <h2
        className="main_header"
      >
        Job Applications
      </h2>
      <div
        className="section2_content"
      >
        <a
          className="add"
          href="/newapp"
          onClick={[Function]}
        >
          <div
            className="squarebutton_content"
          >
            <button
              className="squarebutton"
            >
              +
            </button>
          </div>
        </a>
        <div
          className="filter"
        >
          <label
            htmlFor="filter"
          >
            Filter By: 
          </label>
          <select
            id="filter"
            onChange={[Function]}
          >
            <option>
              All
            </option>
            <option>
              Interested
            </option>
            <option>
              Applied
            </option>
            <option>
              Closed
            </option>
          </select>
        </div>
      </div>
    </section>
  </div>,
  <section
    className="section3"
  >
    <div
      className="container"
    >
      <table
        className="responsive-table"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              Job
            </th>
            <th
              scope="col"
            >
              Company
            </th>
            <th
              scope="col"
            >
              Date Applied
            </th>
            <th
              scope="col"
            >
              Interview Date
            </th>
            <th
              className="last_th"
              scope="col"
            >
              Status
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <p
      className="no_applications"
    >
      Add a new job application to get started!
    </p>
  </section>,
]
`;
